src/modules/DNS.rb

```diff
@@ -239,7 +239,13 @@ def IsHostLocal(check_host)
       # loopback interface or localhost hostname
       return true if ["127.0.0.1", "::1", "localhost", "localhost.localdomain"].include?(check_host)
 
-      ip_addresses = connections.map { |c| c.all_ips.map { |i| i&.address&.address.to_s } }.flatten
+      ip_addresses = []
```

Comment:
you can use here each_with_object with benefits like
```
ip_addresses = connections.each_with_object([]) do |conn, res|
  conn.all_ips.each_with_object(res) do |ip, ips|
     address = ip&.address&.address.to_s
     ips << address if !address.empty? && !ip_addresses.include?(address)
   end
end
```

Reply:
Was not sure about using it, the benefits could depends on the eyes reading it :P do not know if prefer it over the original. But if is fine with me

Reply:
Apart that the connections does not implement the each_with_object method, so we could add to it, or omit by now

Reply:
Adapted anyway